commit 0ad65bd7e1b38dd2c86da5f07dbfa35fe1f03099
Author: Alina Friedrichsen <x-alina@gmx.net>
Date:   Mon Mar 2 23:29:48 2009 +0100

    ath5k: Set TSF fix
    
    The old code doesn't work correctly e.g. on newer chipsets like AR5418+AR2122 and AR5416+AR2133.
    
    Signed-off-by: Alina Friedrichsen <x-alina@gmx.net>
    Signed-off-by: John W. Linville <linville@tuxdriver.com>

diff --git a/drivers/net/wireless/ath5k/pcu.c b/drivers/net/wireless/ath5k/pcu.c
index f8a4a69..55122f1 100644
--- a/drivers/net/wireless/ath5k/pcu.c
+++ b/drivers/net/wireless/ath5k/pcu.c
@@ -657,9 +657,8 @@ void ath5k_hw_set_tsf64(struct ath5k_hw *ah, u64 tsf64)
 {
 	ATH5K_TRACE(ah->ah_sc);
 
-	ath5k_hw_reg_write(ah, 0x00000000, AR5K_TSF_L32);
-	ath5k_hw_reg_write(ah, (tsf64 >> 32) & 0xffffffff, AR5K_TSF_U32);
 	ath5k_hw_reg_write(ah, tsf64 & 0xffffffff, AR5K_TSF_L32);
+	ath5k_hw_reg_write(ah, (tsf64 >> 32) & 0xffffffff, AR5K_TSF_U32);
 }
 
 /**
